core/blockarchiver/client.go

```diff
@@ -0,0 +1,229 @@
+package blockarchiver
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	bundlesdk "github.com/bnb-chain/greenfield-bundle-sdk/bundle"
+)
+
+// Client is a client to interact with the block archiver service
+type Client struct {
+	hc                *http.Client
+	blockArchiverHost string
+	spHost            string
+	bucketName        string
+}
+
+func New(blockAchieverHost, spHost, bucketName string) (*Client, error) {
+	transport := &http.Transport{
+		DisableCompression:  true,
+		MaxIdleConnsPerHost: 1000,
+		MaxConnsPerHost:     1000,
+		IdleConnTimeout:     90 * time.Second,
+	}
+	client := &http.Client{
+		Timeout:   10 * time.Minute,
+		Transport: transport,
+	}
+	return &Client{hc: client, blockArchiverHost: blockAchieverHost, spHost: spHost, bucketName: bucketName}, nil
+}
+
+func (c *Client) GetBlockByHash(ctx context.Context, hash common.Hash) (*Block, error) {
+	payload := preparePayload("eth_getBlockByHash", []interface{}{hash.String(), "true"})
```

Comment:
why not use `ethclient.Client{}.BlockByNumber()` as the `blockAchieverHost` provide exactly compatible API